index.html

```diff
@@ -3989,10 +3989,25 @@ <h3>JSON-LD</h3>
 
         <p>
 As elaborated upon in Section <a href="#json-processing"></a>, some software
-applications might not perform full JSON-LD processing.
-<a>Conforming document</a> authors are urged to not use JSON-LD keywords that
-affect document processing in the `@context` value, such as globally setting
-`@base` and `@language`, in order to increase interoperability.
+applications might not perform full JSON-LD processing. In order to increase
+interoperability, <a>conforming document</a> authors are urged to not use
+JSON-LD features that are not easily detected when not performing full
+JSON-LD processing. This guidance includes:
+        </p>
+
+        <ul>
+          <li>
+Not using JSON-LD keywords in the `@context` value that modify document
+processing, such as globally setting `@base` and `@language`, in order to
+increase interoperability.
+          </li>
+          <li>
+Always using the compact form for JSON-LD terms and types and always ensuring
+that those terms and types are defined in application-specific JSON-LD context
+files.
```

Comment:
Suggest replacing the above three lines by:

Always using the compact form for JSON-LD terms and types when these are defined in JSON-LD context files.
Application specific terms and types MAY use the long (URI) form when the compact forms are not defined in any JSON-LD context file.

Reply:
@David-Chadwick -- I'm confused by your comments. Note that the `Use ... files` in green in my change request, which replaced the three lines you're suggesting be replaced by something else, is an item in unordered list of "avoid using these JSON-LD features" ... and you seem to be saying "don't tell implementers to use JSON-LD context files" which is already happening. Part of my confusion stems, I think, from your having made 3 comments on the same bullet point, which comments don't seem to be fully connected thoughts, nor to take my existing change request into consideration.

My sense is that you're reading the PR Preview, which doesn't reflect suggested changes until/unless the suggestions are merged.... Please take another look at the HTML source, and base your suggestion on the total context of that unordered list, starting with [line 3991](https://github.com/w3c/vc-data-model/pull/1276/files#r1325509056) and continuing to line 4011?

Reply:
> [@David-Chadwick] it appears to only remove the word 'always'

I strongly suggest that you re-examine [my change request for that paragraph (bullet point)](https://github.com/w3c/vc-data-model/pull/1276#discussion_r1325512116), and tell me again whether it appears to only remove the word 'always'. If so, I will have little choice but to ignore your comments, as you are not responding to the full flesh of my own; indeed, you're acting as if my change requests do not exist.

PRs on GitHub are conversations. PR Preview is a *helper* but it only shows the result of the PR as initially drafted, and as/when conversation on the PR results in _revision_ of the PR (which is to revise the text of the document). Ignoring the conversational aspect of this tool does a disservice to all involved, yourself included.

(To be most explicit: My change request significantly rewords the content of that bullet, which is to make it make sense in context of my small rewording of the intro to those bullets, along with my suggested changes to the other bullets in the list, which are to make them all agree with one another. Your own change only applies to the single bullet, not to the intro, nor the other bullets, and does not appear to me to help in bringing clarity to this segment of the document.)

Reply:
@David-Chadwick wrote:
> Always using the compact form for JSON-LD terms and types when these are defined in JSON-LD context files.

This seems fine.

> Application specific terms and types MAY use the long (URI) form when the compact forms are not defined in any JSON-LD context file.

Hrm, but we have a base `@vocab` that makes it so they don't have to do that. I believe there were objections in the group to using long-form URLs and adding the text you suggest above would re-raise those objections. @OR13, I believe you were one of the people that wanted clear guidance here -- thoughts?

Reply:
Sorry, I escaped the thread: https://github.com/w3c/vc-data-model/pull/1276#issuecomment-1739187061

The linked comment is probably a better way to address this issue.... the default behavior should not appear to be illegal, imo.

Reply:
`Application specific terms and types MAY use the long (URI) form when the compact forms are not defined in any JSON-LD context file.`
The reason for adding this text is that it still allows compact local names to be used because of the `@vocab` term, but it also allows implementors who wish to use a global long (URI) name to use that as well, ensuring that no other VC Issuer will have a name clash with their invented compact local name when they do not want to start creating their own `@context` files

Reply:
I've attempted to address this issue in the latest series of edits, please re-review @David-Chadwick.